lib/logical_solver/presolve.cpp

```diff
@@ -0,0 +1,889 @@
+#include "presolve.hpp"
+#include <iostream>
+
+namespace logical_solver{
+
+  Presolve::Presolve(
+    const std::vector<std::vector<int>> problem_matrix, 
+    const std::vector<int> lower_bounds, 
+    const std::vector<int> upper_bounds,
+    const int inequalities_count,
+    const int equalities_count,
+    const bool solve_MIP
+  ) {
+    // Infitnity - eventually to be defined in types.
+    kInfinity = std::numeric_limits<int>::max(); 
```

Comment:
But definitely shouldn't be here since it's assigned each time the class is instantiated. I know this is a single, trivial operation but this class is instantiated 150k times. That's 150k pointless operations...

Reply:
That should be done now.